packages/commcare/src/Adaptor.js

```diff
@@ -77,23 +78,62 @@ export function get(path, params = {}, callback = s => s) {
       params
     );
 
+    let { offset = 0, limit = 1000 } = resolvedParams;
+
+    let nextState = state;
+    let result;
+    let allowPagination = !isNaN(resolvedParams.offset);
+
     try {
-      const response = await request(
-        state.configuration,
-        `/a/${domain}/api/v0.5/${resolvedPath}`,
-        {
-          method: 'GET',
-          params: resolvedParams,
-          contentType: 'application/json',
+      let requestParams = {
+        ...resolvedParams,
+      };
+
+      do {
+        const response = await request(
+          state.configuration,
+          `/a/${domain}/api/v0.5/${resolvedPath}`,
+          {
+            method: 'GET',
+            params: requestParams,
+            contentType: 'application/json',
+          }
+        );
+
+        nextState = prepareNextState(state, response, callback);
+
+        if (response?.body?.meta?.next) {
+          if (!result) {
+            result = [];
+          }
+          offset = response?.body?.meta?.offset + response?.body?.meta?.limit;
+          limit = response?.body?.meta?.limit;
+
+          requestParams = {
+            ...requestParams,
+            offset,
+            limit,
+          };
+          result.push(...nextState.data);
+        } else {
+          if (result) {
+            result.push(...nextState.data);
+          } else {
+            result = nextState.data;
+          }
+
+          break;
         }
-      );
-
-      return prepareNextState(state, response, callback);
+      } while (allowPagination);
+      return {
```

Comment:
I would like to see a clear  line break between the `while` and the `return`.

The return is really important but because the code is too dense, it's a little hard to read.